lib/include/openamp/rpmsg.h

```diff
@@ -73,6 +74,12 @@ struct rpmsg_endpoint {
 	/** Address of the default remote endpoint binded */
 	uint32_t dest_addr;
 
+	/** Reference count of the endpoint */
+	uint32_t refcnt;
+
+	/** Callback to free the endpoint */
```

Comment:
Callback to inform the user that the endpoint allocation can be safely removed (optional).